datafusion/proto/src/physical_plan/from_proto.rs

```diff
@@ -560,9 +560,21 @@ pub fn parse_protobuf_file_scan_config(
         output_ordering.push(sort_expr);
     }
 
+    // Currently when the schema for the file is set the partition columns
+    // are present, which is illegal because they aren't actually in the files.
+    // This is a workaround to remove them from the schema.
+    let file_schema = Arc::new(Schema::new(
```

Comment:
I don't undersrtand why the protobuf code would be doing this filtering -- wouldn't this potentially mask an error in whatever was creating the protobuf?

Perhaps it would be better to apply this filtering prior to serializing the plan?

If this is needed in the protobuf serialization code, can you please add a test to 
1. Show how this happens and 
2. Ensure we don't accidentally break this functionality in a future refactor?

Reply:
Absolutely.

Reply:
These are complete.